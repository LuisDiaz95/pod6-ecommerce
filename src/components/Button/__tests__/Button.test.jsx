import Button from '../Button';
import { render, screen, fireEvent } from '@testing-library/react';
//aaa
describe('<Button/>', () => {
  test('Should has the correct class names when is marked as primary button', () => {
    //arrange
    render(<Button children="Something" variant="primary" />);

    //act
    let btn = screen.getByText("Something");

    //assert
    expect(btn.className).toBe(" Button btn--primary ")
  });

  test('Should has the correct class names when is marked as secondary button', () => {
    render(<Button children="Something" variant="secondary" />);

    let btn = screen.getByText("Something");

    expect(btn.className).toBe(" Button btn--secondary ")
  });

  test('Should propagate the click event to its parent', () => {
    let spy = jest.fn();
    render(<Button children="Something" variant="secondary" onClick={spy} />);
    
    let btn = screen.getByText("Something");

    fireEvent.click(btn);

    expect(spy).toHaveBeenCalled();
  });

  test('Should change the state of a button to disabled', () => {
    render(<Button children="Something" variant="secondary" disabled={true} />);

    let btn = screen.getByText("Something");

    expect(btn).toMatchSnapshot('Check disabled status');
  });

  test('Should change the text inside the button', () => {
    let { rerender } = render(<Button children="Something" variant="primary" />);
 
    let btn = screen.getByText("Something");

    rerender(<Button children={"Something else"} variant="primary" />);

    expect(btn.textContent).toBe("Something else");
  });

});