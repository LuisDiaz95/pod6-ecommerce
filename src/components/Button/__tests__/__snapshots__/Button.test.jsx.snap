// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Should change the state of a button to disabled: Check disabled status 1`] = `
<button
  class=" Button btn--secondary "
  disabled=""
>
  Something
</button>
`;
